examples/scripts/chat.py

```diff
@@ -0,0 +1,302 @@
+        filename = create_default_filename(args.model)
+        filename = os.path.join(folder, filename)
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+
+    with open(filename, "w") as f:
+        json.dump(output_dict, f, indent=4)
+    return os.path.abspath(filename)
+
+def clear_chat_history(system_prompt):
+    if system_prompt is None:
+        chat = []
+    else:
+        chat = [{"role": "system", "content": system_prompt}]
+    return chat
+
+def parse_settings(user_input, current_args, interface):
+    settings = user_input[4:].strip().split(";")
+    settings = [(setting.split("=")[0], setting[len(setting.split("=")[0])+1:]) for setting in settings]
+    settings = dict(settings)
+    error = False
+
+    for name in settings:
+        if hasattr(current_args, name):
+            
+            try:
+                if isinstance(getattr(current_args, name), bool):
+                    if settings[name] == "True":
+                        settings[name] = True
+                    elif settings[name] == "False":
+                        settings[name] = False
+                    else:
+                        raise ValueError
+                else:
+                    settings[name] = type(getattr(current_args, name))(settings[name])
+            except ValueError:
+                interface.print_red(f"Cannot cast setting {name} (={settings[name]}) to {type(getattr(current_args, name))}.")
+        else: 
+            interface.print_red(f"There is no '{name}' setting.")
+    
+    if error:
+        interface.print_red("There was an issue parsing the settings. No settings have been changed.")
+        return current_args, False
+    else:
+        for name in settings:
+            setattr(current_args, name, settings[name])
+            interface.print_green(f"Set {name} to {settings[name]}.")
+
+        time.sleep(1.5) # so the user has time to read the changes
+        return current_args, True
+
+
+def load_model(args):
+    torch_dtype = (
+        args.torch_dtype
+        if args.torch_dtype in ["auto", None]
+        else getattr(torch, args.torch_dtype)
+    )
+    quantization_config = get_quantization_config(args)
+    model_kwargs = dict(
+        revision=args.model_revision,
+        trust_remote_code=args.trust_remote_code,
+        attn_implementation=args.attn_implementation,
+        torch_dtype=torch_dtype,
+        device_map=get_kbit_device_map() if quantization_config is not None else None,
+        quantization_config=quantization_config,
+    )
+    model = AutoModelForCausalLM.from_pretrained(args.model, **model_kwargs)
+
+    if getattr(model, "hf_device_map", None)  is None:
+        model = model.to(args.device)
+    print(model.device)
+    return model
+
+def chat_cli():
+    parser = HfArgumentParser((ChatArguments))
+    args = parser.parse_args()
+
+    current_args = copy.deepcopy(args)
+
+    with open(os.path.join(os.path.dirname(__file__), 'default_chat_config.yaml'), "r") as f:
```

Comment:
why not using the `TrlParser` above and use `parser.parse_args_and_config`: https://github.com/huggingface/trl/blob/6cfa5cfc817be0f8acf30f8158a4a0ea244dc988/examples/scripts/sft.py#L85 ?